tests/jobs/python/shuffle_app.py

```diff
@@ -0,0 +1,40 @@
+from __future__ import print_function
+import sys
+from random import getrandbits
+from time import sleep
+
+from pyspark.sql import SparkSession
+
+# ShuffleApp test job ported from Scala to Python
```

Comment:
While this job serves as a good smoke test - it doesn't verify Scala 2.12 support. Let's instead build two distributions of our tests jar: 2.11 and 2.12 and expose Scala version as a parameter so we can test Spark 2.12 with integration tests jar 2.12 and Spark 2.11 with 2.11